plugins/processors/awsappsignals/internal/resolver/eks.go

```diff
@@ -0,0 +1,739 @@
+		} else if len(workloads) == 1 {
+			m.serviceToWorkload.Store(serviceAndNamespace, workloads[0])
+		} else {
+			m.logger.Debug("No workload found for service", zap.String("service", serviceAndNamespace))
+			m.deleter.DeleteWithDelay(m.serviceToWorkload, serviceAndNamespace)
+		}
+		return true
+	})
+}
+
+func (m *ServiceToWorkloadMapper) Start(stopCh chan struct{}) {
+	// do the first mapping immediately
+	m.MapServiceToWorkload()
+	m.logger.Debug("First-time map service to workload at:", zap.Time("time", time.Now()))
+
+	go func() {
+		for {
+			select {
+			case <-stopCh:
+				return
+			case <-time.After(time.Minute + 30*time.Second):
+				m.MapServiceToWorkload()
+				m.logger.Debug("Map service to workload at:", zap.Time("time", time.Now()))
+			}
+		}
+	}()
+}
+
+func getEksResolver(logger *zap.Logger) subResolver {
+	once.Do(func() {
+		config, err := clientcmd.BuildConfigFromFlags("", "")
+		if err != nil {
+			logger.Fatal("Failed to create config", zap.Error(err))
+		}
+
+		clientset, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			logger.Fatal("Failed to create eks client", zap.Error(err))
+		}
+
+		// add a time jitter of 10 seconds
+		jitterSleep(10)
+
+		sharedInformerFactory := informers.NewSharedInformerFactory(clientset, 0)
+		podInformer := sharedInformerFactory.Core().V1().Pods().Informer()
+		serviceInformer := sharedInformerFactory.Core().V1().Services().Informer()
+
+		timedDeleter := &TimedDeleter{Delay: deletionDelay}
+		podWatcher := NewPodWatcher(logger, podInformer, timedDeleter)
+		serviceWatcher := NewServiceWatcher(logger, serviceInformer, timedDeleter)
+
+		safeStopCh := &safeChannel{ch: make(chan struct{}), closed: false}
+		// initialize the pod and service watchers for the cluster
+		podWatcher.Run(safeStopCh.ch)
+		serviceWatcher.Run(safeStopCh.ch)
+		// wait for caches to sync (for once) so that clients knows about the pods and services in the cluster
+		podWatcher.WaitForCacheSync(safeStopCh.ch)
+		serviceWatcher.WaitForCacheSync(safeStopCh.ch)
+
+		serviceToWorkload := &sync.Map{}
+		serviceToWorkloadMapper := NewServiceToWorkloadMapper(serviceWatcher.serviceAndNamespaceToSelectors, podWatcher.workloadAndNamespaceToLabels, serviceToWorkload, logger, timedDeleter)
+		serviceToWorkloadMapper.Start(safeStopCh.ch)
+
+		instance = &eksResolver{
+			logger:                         logger,
+			clientset:                      clientset,
+			ipToServiceAndNamespace:        serviceWatcher.ipToServiceAndNamespace,
+			serviceAndNamespaceToSelectors: serviceWatcher.serviceAndNamespaceToSelectors,
+			ipToPod:                        podWatcher.ipToPod,
+			podToWorkloadAndNamespace:      podWatcher.podToWorkloadAndNamespace,
+			workloadAndNamespaceToLabels:   podWatcher.workloadAndNamespaceToLabels,
+			serviceToWorkload:              serviceToWorkload,
+			workloadPodCount:               podWatcher.workloadPodCount,
+			safeStopCh:                     safeStopCh,
+		}
+
+		go instance.debugPrint()
```

Comment:
Should this be enabled by default?

Reply:
probably not. @thpierce / @nanzhenAWS can we remove this and implement debug lines as part of the logger instead?

Reply:
Yes

Reply:
This debug print is to give a full picture about the various mappings for debug purposes. This should be disabled by default.